embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt

```diff
@@ -29,24 +29,13 @@ internal class EmbraceBreadcrumbService(
     private val clock: Clock,
     private val configService: ConfigService,
     private val activityTracker: ActivityTracker,
-    sessionSpanWriter: SessionSpanWriter,
-    spanService: SpanService,
-    private val logger: InternalEmbraceLogger
+    private val dataSourceModuleProvider: Provider<DataSourceModule?>,
```

Comment:
As discussed, because you reordered the module init, you shouldn't need a provider here. But lets merge it and we can fix it later.